src/Microsoft.Sbom.Api/Utils/PackageDetailsFactory.cs

```diff
@@ -0,0 +1,256 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Xml;
+using Microsoft.ComponentDetection.Contracts.BcdeModels;
+using Microsoft.Sbom.Api.Output.Telemetry;
+using Microsoft.Sbom.Api.Utils;
+using Microsoft.Sbom.Common;
+using NuGet.Configuration;
+using Serilog;
+
+namespace Microsoft.Sbom.Api.Executors;
+
+public class PackageDetailsFactory : IPackageDetailsFactory
+{
+    private readonly IFileSystemUtils fileSystemUtils;
+    private readonly ILogger log;
+    private readonly IRecorder recorder;
+
+    public PackageDetailsFactory(IFileSystemUtils fileSystemUtils, ILogger log, IRecorder recorder)
+    {
+        this.fileSystemUtils = fileSystemUtils ?? throw new ArgumentNullException(nameof(fileSystemUtils));
+        this.log = log ?? throw new ArgumentNullException(nameof(log));
+        this.recorder = recorder ?? throw new ArgumentNullException(nameof(recorder));
+    }
+
+    public ConcurrentDictionary<(string, string), PackageDetailsObject> GetPackageDetailsDictionary(IEnumerable<ScannedComponent> scannedComponents)
+    {
+        var packageDetailsLocations = GetPackageDetailsLocations(scannedComponents);
+
+        return ExtractPackageDetailsFromFiles(packageDetailsLocations);
+    }
+
+    private List<string> GetPackageDetailsLocations(IEnumerable<ScannedComponent> scannedComponents)
+    {
+        var packageDetailsConfirmedLocations = new List<string>();
+
+        foreach (var scannedComponent in scannedComponents)
+        {
+            var componentType = scannedComponent.Component.PackageUrl?.Type.ToLower();
```

Comment:
No need to go via `PackageUrl`

```suggestion
            var componentType = scannedComponent.Component.Type;
```